docs/proposals/prevent-removal-managed-resources/README.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Prevent removal managed resources
+authors:
+- "@CharlesQQ"
+reviewers:
+- "@RainbowMango"
+- "@XiShanYongYe-Chang"
+- "@chaunceyjiang"
+approvers:
+
+creation-date: 2024-06-27
+
+# prevent removal managed resource
+
+## Summary
+
+A deletion mechanism and its API is provided to prevent cascade resource in member cluster deleted by control plane.
+
+## Motivation
+
+All resources reconciled by the execution controller have a finalizer `karmada.io/execution-controller` added to work object's metedata.
+This finalizer will prevent deletion of work object until the execution controller has a chance to perform pre-deletion cleanup.
+
+However, in some cases, user want to delete resource in control plane but resource in member cluster kept, like `kubectl delete sts xxx --cascade=orphan`, sts is deleted but pods not.
+karmada have different types of resources, different resource might have different prevent removal process.
+
+1. resource template: cleanup the labels/annotations from the resource on member clusters and delete work object.
+2. dependencies resource: cleanup the labels/annotations from the dependencies resource on member clusters,such as configmap/secret, delete work object.
+3. directly generate work object such as namespace/cronfederatedhpa/federaredhpa/federatedresourcequota execute the same strategy as resource template.
+
+### Goals
+- Provides API to prevent delete cascade resource.
+- Provides command line to use feature like `karmadactl orphaningdeletion complete/disable <resource type> <name>`
```

Comment:
do you mean that the command `karmadactl orphaningdeletion complete` will only patch such an annotaion `resourcetemplate.karmada.io/deletionpolicy=complete` to resource templates?
if so, does this command seem too useless? can you provide such a command `karmadactl delete --cascade=orphan`?